### PR TITLE
replaced sdm with CDD.

### PR DESCRIPTION
**Pull request by nshehz**

### CODE REVIEW

File: tests/executables_src/testRebotConnectionTimeouts.cpp [hunk 1]
```diff
@@ -56,8 +56,10 @@ BOOST_FIXTURE_TEST_CASE(testOpenConnection, F) {
 BOOST_FIXTURE_TEST_CASE(testReadTimeout, F) {
   uint32_t timeout_sec = 1;
   auto accetable_completion_time = std::chrono::seconds(timeout_sec * 5);
-  Device d("sdm://./rebot=localhost," + std::to_string(rebotServer.port()) + ",mtcadummy_rebot.map," +
-      std::to_string(timeout_sec));
+  Device d("(rebot?ip=localhost&port=" + std::to_string(rebotServer.port()) + "&map=mtcadummy_rebot.map&timeout=" +
+      std::to_string(timeout_sec)+")");
+
+  //Device d("rebot?ip=localhost&port=5001&map=mtcadummy_rebot.map");
```
Comment (killenb) [inline tests/executables_src/testRebotConnectionTimeouts.cpp:62]: Coding style: Don't leave commented code lines.

File: tests/unitTestsNotUnderCtest/testRebotBackend.cpp [hunk 2]
```diff
@@ -102,11 +102,11 @@ DeviceInfo RebotTestClass::getDeviceDetailsFromDMap(const std::string& cardAlias
 }
 
 RebotServerDetails RebotTestClass::extractServerDetailsFromUri(std::string& uri) {
-  ChimeraTK::Sdm parsedSDM = ChimeraTK::Utilities::parseSdm(uri);
-  std::list<std::string>& serverParameters = parsedSDM._Parameters;
-  std::list<std::string>::iterator it = serverParameters.begin();
-  std::string ip = *it;
-  std::string port = *++it;
+  std::cout<<uri<<std::endl;
```
Comment (killenb) [inline tests/unitTestsNotUnderCtest/testRebotBackend.cpp:105]: remove leftover debug output

File: tests/executables_src/testBackendFactory.cpp [hunk 3]
```diff
@@ -80,45 +80,45 @@ BOOST_AUTO_TEST_CASE(testPluginMechanism) {
...
-  BOOST_CHECK_THROW(BackendFactory::getInstance().createBackend("sdm://./wrongVersionBackend=goodMapFile.map"),
+  BOOST_CHECK_THROW(BackendFactory::getInstance().createBackend("(wrongVersionBackend?map=goodMapFile.map)"),
       ChimeraTK::logic_error);
 
   BOOST_CHECK_NO_THROW(ChimeraTK::BackendFactory::getInstance().loadPluginLibrary("./libWrongVersionBackendCompat.so"));
-  BOOST_CHECK_THROW(BackendFactory::getInstance().createBackend("sdm://./libWrongVersionBackendCompat=goodMapFile.map"),
+  BOOST_CHECK_THROW(BackendFactory::getInstance().createBackend("(ibWrongVersionBackendCompat?map=goodMapFile.map)"),
       ChimeraTK::logic_error);
 
+  //BOOST_CHECK_THROW(BackendFactory::getInstance().createBackend("(unregisteredBackend)"), ChimeraTK::logic_error);
```
Comment (killenb) [inline tests/executables_src/testBackendFactory.cpp:112]: coding style: remove commented code lines

**Review by killenb — CHANGES_REQUESTED** [review]
tests/executables_src/testBackendFactory.cpp
tests/executables_src/testRebotConnectionTimeouts.cpp
tests/unitTestsNotUnderCtest/testRebotBackend.cpp
tests/unitTestsNotUnderCtest/testRebotBackendCreation.cpp

are not passing the coding style test. Running clang-format on them introduces some changes (white space/line breaks) on my system (which are correct as far as I can see). We have to check why this is not working for you.

**Comment by killenb** [comment]
Please also update to master with merge commit (if you rebase the review will lose the commits it is referring to)

File: tests/dMapDir/testRelativePaths.dmap [hunk 4]
```diff
@@ -1,9 +1,10 @@
...
-DUMMYD1   sdm://./dummy=mapFile.map mapFile.map
+DUMMYD0   (dummy?map=../mtcadummy_withoutModules.map)
+DUMMYD0   (dummy?map=../mtcadummy_withoutModules.map)
+DUMMYD1   (dummy?map=mapFile.map)
 
-example0   sdm://./exx:examp=bus,1  ../mtcadummy_withoutModules.map
-example1   sdm://./exx:examp=bus,1  ./mapFile.map
\ No newline at end of file
+example0  (exx:examp?parameter=bus,parameter2=1?map=../mtcadummy_withoutModules.map)
+example1  (exx:examp?parameter=bus,parameter2=1?map=./mapFile.map)
```
[the remark this answers is not in the document]
Comment (nshehz) [reply tests/dMapDir/testRelativePaths.dmap:10]: Woops. Seems my replace find did not work well.Sorry

File: tests/dummies.dmap [hunk 5]
```diff
@@ -1,21 +1,20 @@
...
+mskrebot1 (rebot?ip=localhost&port=5001&map=mtcadummy_rebot.map)
 #an entry with the old /dev/ syntax for compatibility tests
 OLD_PCIE  /dev/mtcadummys0 mtcadummy.map
 SEQUENCES   (dummy?map=sequences.map)
 INVALID_SEQUENCES   (dummy?map=invalidSequences.map)
-PCIE_DOUBLEMAP sdm://./pci:pcieunidummys6=mtcadummy.map mtcadummy.map
-REBOT_DOUBLEMAP sdm://./rebot=localhost,5001,mtcadummy_rebot.map mtcadummy_rebot.map
-
+PCIE_DOUBLEMAP (pci:pcieunidummys6?map=mtcadummy.map)
+REBOT_DOUBLEMAP (rebot?ip=localhost&port=5001&map=mtcadummy_rebot.map)
```
[the remark this answers is not in the document]
Comment (nshehz) [reply tests/dummies.dmap:20]: Really. hmm Rebot backend looks for 4 parameters in map.

File: tests/executables_src/testRebotConnectionTimeouts.cpp (hunk 1 above)
Comment (nshehz) [reply tests/executables_src/testRebotConnectionTimeouts.cpp:62]: Why I do not see them in Jenkins branch :(

File: tests/executables_src/testBackendFactory.cpp (hunk 3 above)
Comment (nshehz) [reply tests/executables_src/testBackendFactory.cpp:112]: my bad.

File: tests/dummies.dmap (hunk 5 above)
[the remark this answers is not in the document]
Comment (killenb) [reply tests/dummies.dmap:19]: With third parameter I mean the third column in the dmap file.
The whole CDD is the second parameter in the map file. The CDD itself then can have as many parameters as it needs (up to 4 in the case of ReboT).

File: tests/executables_src/testRebotConnectionTimeouts.cpp (hunk 1 above)
Comment (killenb) [reply tests/executables_src/testRebotConnectionTimeouts.cpp]: This is not part of the unit test and automatic code formatting. The formatter cannot  know if it is a real comment or a  line of code which has been commented out during development.
It is the same category as "remove debug cout". It's valid C++, so the automatic tools cannot find this.

**Comment by nshehz** [comment]
Clag formatting is fixed. 
coding_style test is passing properly now.

**Review by killenb — APPROVED** [review]

**Pull request merged**